Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp

```diff
@@ -540,10 +555,19 @@ namespace ScriptAutomation
             ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(operation));
         }
 
-        void CompareScreenshots(const AZStd::string& filePathA, const AZStd::string& filePathB, float minDiffFilter)
+        void CompareScreenshots(const AZStd::string& compareName, const AZStd::string& comparisonLevel, const AZStd::string& filePathA, const AZStd::string& filePathB, float minDiffFilter)
         {
-            auto operation = [filePathA, filePathB, minDiffFilter]()
+            // capture strings by copy or risk them being deleted before we access them.
+            auto operation = [=]()
             {
+
+                const ImageComparisonToleranceLevel* toleranceLevelPtr = ScriptAutomationInterface::Get()->FindToleranceLevel(comparisonLevel);
+                if (!toleranceLevelPtr)
+                {
+                    AZ_Error("ScriptAutomation", false, "Failed to find image comparison level named %s", comparisonLevel.c_str());
+                    return;
+                }
+                const ImageComparisonToleranceLevel toleranceLevel = *toleranceLevelPtr; // duplicate in case data is reloaded
```

Comment:
This comment doesn't make sense to me, isn't it just dereffing a pointer now that you know it's not null?

Reply:
My point was just that the operation can be happening on a different thread than the main thread, so it's possible for the pointer to be removed while we are trying to use it.

Reply:
Updated to remove the copy deref to local instance.